### PR TITLE
New Package: xfce4-terminal - Xfce libs added as well

### PR DESCRIPTION
**Pull request by ThatGeekyWeeb**

### CODE REVIEW

File: packages/libxfce4ui.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+require 'package'
+
+class Libxfce4ui < Package
+    description 'Replacement of the old libxfcegui4 library'
+    homepage "https://xfce.org/"
+    version '4.15.3'
+    compatibility 'all'
+    source_url "https://archive.xfce.org/src/xfce/libxfce4ui/4.15/libxfce4ui-#{version}.tar.bz2"
```
Comment (uberhacker) [inline packages/libxfce4ui.rb:8]: I prefer to not use `#{version}` in urls.  It makes it easier to `sha256sum` and cut and paste.

File: packages/libxfce4util.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+require 'package'
+
+class Libxfce4util < Package
+    description 'Utility library for the Xfce4 desktop environment'
+    homepage "https://xfce.org/"
+    version '4.14'
```
Comment (uberhacker) [inline packages/libxfce4util.rb:6]: version '4.14.0' ?

File: packages/libxfce4util.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    compatibility 'all'
+    source_url "https://archive.xfce.org/src/xfce/libxfce4util/4.14/libxfce4util-#{version}.0.tar.bz2"
```
Comment (uberhacker) [inline packages/libxfce4util.rb:8]: I prefer to not use `#{version}` in urls.  It makes it easier to `sha256sum` and cut and paste.

File: packages/xfce4_dev_tools.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+require 'package'
+
+class Xfce4_dev_tools < Package
+    description 'Xfce4 development tools'
+    homepage "https://xfce.org/"
+    version '4.14'
+    compatibility 'all'
+    source_url "https://archive.xfce.org/src/xfce/xfce4-dev-tools/#{version}/xfce4-dev-tools-#{version}.0.tar.bz2"
```
Comment (uberhacker) [inline packages/xfce4_dev_tools.rb:8]: I prefer to not use `#{version}` in urls.  It makes it easier to `sha256sum` and cut and paste.

File: packages/xfce4_dev_tools.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+require 'package'
+
+class Xfce4_dev_tools < Package
+    description 'Xfce4 development tools'
+    homepage "https://xfce.org/"
+    version '4.14'
```
Comment (uberhacker) [inline packages/xfce4_dev_tools.rb:6]: version '4.14.0' ?

**Comment by uberhacker** [comment]
Thank you for the new packages @ThatGeekyWeeb!  Please see comments.

**Comment by ThatGeekyWeeb** [comment]
I've made the requested changes and added the `exo` dep I had originally placed on `xfce4_terminal.rb`, which I had packaged as well, - Which is needed , and missing atm.

File: packages/exo.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+class Exo < Package
+    description 'Extension library for the Xfce desktop environment'
+    homepage "https://xfce.org/"
+    version '0.12.11'
+    compatibility 'all'
+    source_url "https://archive.xfce.org/src/xfce/exo/0.12/exo-0.12.11.tar.bz2"
+    source_sha256 'ec892519c08a67f3e0a1f0f8d43446e26871183e5aa6be7f82e214f388d1e5b6'
+    
+    depends_on 'libtool'
+    depends_on 'automake'
```
Comment (uberhacker) [inline packages/exo.rb:12]: libtool and automake are core packages and don't need to be listed here.

File: packages/libxfce4util.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+class Libxfce4util < Package
+    description 'Utility library for the Xfce4 desktop environment'
+    homepage "https://xfce.org/"
+    version '4.14.0'
+    compatibility 'all'
+    source_url "https://archive.xfce.org/src/xfce/libxfce4util/4.14/libxfce4util-4.14.0.tar.bz2"
+    source_sha256 '32ad79b7992ec3fd863e8ff2f03eebda8740363ef9d7d910a35963ac1c1a6324'
+    
+    depends_on 'intltool' => :build
+    depends_on 'python3' # Must be version 3.8 or else it WILL fail! - There are some issues with 3.8.2 - I had 3.6
```
Comment (uberhacker) [inline packages/libxfce4util.rb:12]: intltool and python3 are core packages and don't need to be listed here.

File: packages/xfce4_dev_tools.rb [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+class Xfce4_dev_tools < Package
+    description 'Xfce4 development tools'
+    homepage "https://xfce.org/"
+    version '4.14.0'
+    compatibility 'all'
+    source_url "https://archive.xfce.org/src/xfce/xfce4-dev-tools/4.14/xfce4-dev-tools-4.14.0.tar.bz2"
+    source_sha256 '2c9eb8e0fe23e47dc31411a93b683fd1b7a49140e9163f0aab9e94a3d8a0b5fd'
+    
+    depends_on 'libtool'
+    depends_on 'automake'
```
Comment (uberhacker) [inline packages/xfce4_dev_tools.rb:12]: These core packages don't need to be listed here.

File: packages/xfce4_terminal.rb [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    homepage "https://xfce.org/"
+    version '0.8.9'
+    compatibility 'all'
+    source_url "https://archive.xfce.org/src/apps/xfce4-terminal/0.8/xfce4-terminal-0.8.9.tar.bz2"
+    source_sha256 '247683a51a964cfaa6b1e92030afe9f782efebfcb550a464170b53eb94216795'
+    
+    depends_on 'desktop_file_utilities'
+    depends_on 'gtkvte' => :build
+    depends_on 'exo' => :build
+    depends_on 'hicolor_icon_theme'
```
Comment (uberhacker) [inline packages/xfce4_terminal.rb:14]: Does this package work without sommelier?  If not, then you will need `depends_on 'sommelier'`.

File: packages/xfce4_dev_tools.rb [hunk 10]
```diff
@@ -0,0 +1,18 @@
+require 'package'
+
+class Xfce4_dev_tools < Package
+    description 'Xfce4 development tools'
+    homepage "https://xfce.org/"
+    version '4.14.0'
+    compatibility 'all'
+    source_url "https://archive.xfce.org/src/xfce/xfce4-dev-tools/4.14/xfce4-dev-tools-4.14.0.tar.bz2"
+    source_sha256 '2c9eb8e0fe23e47dc31411a93b683fd1b7a49140e9163f0aab9e94a3d8a0b5fd'
+
```
Comment (uberhacker) [inline packages/xfce4_dev_tools.rb:10]: No dependencies?

**Comment by uberhacker** [comment]
FYI, you can find all the core packages [here](https://github.com/skycocker/chromebrew/blob/master/tools/core_packages.txt).

File: packages/xfce4_terminal.rb (hunk 9 above)
Comment (ThatGeekyWeeb) [reply packages/xfce4_terminal.rb:14]: Last I checked it was able to start inside of Aura (Native WM), it complains about lack of a WM but starts none the less!

Comment (ThatGeekyWeeb) [reply packages/xfce4_terminal.rb:14]: I checked again, and that is true! it works even without 'sommelier' installed, and can become transparent within Aura!

Comment (uberhacker) [reply packages/xfce4_terminal.rb:14]: Nice!

**Comment by uberhacker** [comment]
Can we use 2 space indentation instead of 4?

**Comment by ThatGeekyWeeb** [comment]
Sure, I can edit it for that! My text editor defaults to 4 is all.

**Pull request merged**